platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/suppliers/DynamicRevisionRemoteResolveWithMetadataSupplierIntegrationTest.groovy

```diff
@@ -1054,6 +1054,9 @@ group:projectB:2.2;release
         }
 
         when:
+        // bouncycastle has .properties files that are actually binary files
+        // see https://github.com/bcgit/bc-java/commit/0aacc38aefe7e79a6d9cca76bd690c24671c3feb
+        executer.withStackTraceChecksDisabled()
```

Comment:
🤔 This looks like something we might want to fix in a different way.  Do we need the .bin.properties inside the gradle-api jar?  I think if we leave this it means nearly every build with build logic will emit these exceptions at DEBUG. 

@ghale At DEBUG, we print this stacktrace:
https://ge.gradle.org/s/oytbmatbp5sam/tests/task/:dependency-management:embeddedIntegTest/details/org.gradle.integtests.resolve.suppliers.DynamicRevisionRemoteResolveWithMetadataSupplierIvyWithGradleMetadataIntegrationTest/can%20cache%20the%20result%20of%20processing%20a%20rule%20across%20projects%20%5Bwith%20Gradle%20metadata%2C%20Ivy%20repository%5D?focused-exception-line=0-1-969&page=eyIwLTEiOjV9&top-execution=1

This is because the gradle-api jar has a file that ends with `.bin.properties`.  The contents is some kind of properties file, but the format of it is gzipped binary, so `Properties.load(...)` object doesn't work. 

IIUC, we handle this situation by failing to hash just the properties in the file and then hashing it as a blob. Is that correct?

Reply:
>  Do we need the .bin.properties inside the gradle-api jar?

I think we can't just exclude those files since they are needed for the Picnic PQC algorithm. Even if we move them out of gradle-api to some other distribution jar, they will still be analyzed, right?

Reply:
One thing I can do is open a PR for BC to fix the file extension, but I'm not sure if it's the best way to address this.

Reply:
I don't think that's necessary.  

I talked to Gary about this a bit more and he pointed out that the exception that comes out only happens the first time we fingerprint the gradle-api jar, so it's not a performance issue.